app/src/main/java/upday/droidconmvvm/MainActivity.java

```diff
@@ -1,13 +1,121 @@
 package upday.droidconmvvm;
 
-import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import java.util.List;
+
+import rx.android.schedulers.AndroidSchedulers;
+import rx.schedulers.Schedulers;
+import rx.subscriptions.CompositeSubscription;
+import upday.droidconmvvm.datamodel.IDataModel;
+import upday.droidconmvvm.model.Language;
 
 public class MainActivity extends AppCompatActivity {
 
+    @NonNull
+    private CompositeSubscription mSubscription;
+
+    @NonNull
+    private MainViewModel mViewModel;
+
+    @Nullable
+    private TextView mGreetingView;
+
+    @Nullable
+    private Spinner mLanguagesSpinner;
+
+    @Nullable
+    private LanguageSpinnerAdapter mLanguageSpinnerAdapter;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        mViewModel = new MainViewModel(getDataModel());
+        setupViews();
+    }
+
+    private void setupViews() {
+        mGreetingView = (TextView) findViewById(R.id.greeting);
+
+        mLanguagesSpinner = (Spinner) findViewById(R.id.languages);
+        assert mLanguagesSpinner != null;
+        mLanguagesSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
```

Comment:
If you want to show the power or Rx, you could make this listener using Observable.create.
But if you do want to make it as simple as possible, then you can leave it like this